Adds random light flicker

tetraphobia opened the pull request:
:cl:
+Increases station spookiness level
/:cl:

Commit:
Adds random light flicker

Commit:
Update lighting.dm

MrFroztee commented:
Let's christen this as our first PR in what must be seven years.
![gnarlymusicalcat](https://user-images.githubusercontent.com/18686822/30360651-584a60f4-9818-11e7-8eb1-43b12c2a8577.png)

SheepFighter commented:
One would not think that so simple a change could crash Dream Daemon on start up, but here we are...

Time for me to do some tinkering...

SheepFighter commented:
The good news is that I got it working.
The bad newes is that the flickering happens with comical alacrity.
It's like a _Poltergeist_ parody.

SheepFighter committed:
Update lighting.dm

SheepFighter commented on code/modules/power/lighting.dm:
```diff
@@ -395,6 +397,12 @@
 
+// random flicker
+/obj/machinery/light/process()
+	..()
```
Bad idea, forgetting this.

FlattestGuitar commented on code/modules/power/lighting.dm:
```diff
@@ -395,6 +397,12 @@
+// random flicker
+/obj/machinery/light/process()
+	..()
+	if(prob(3))
```
A single `flicker()` call takes on average as much time as its argument - in this case it's 20 seconds.

So, every 2 seconds you have a 3% chance for each and every light on the station to start flickering for 20 seconds.

That means that every light at all times has a `1 - (97%)^10 ~= 10%` chance to be flickering in any given moment.

Tone that down to 1%, you don't want a rave, you want a space station.

Commit:
Fixes typo and nerfs rave

It's meant to flicker twice, not 20 times. Whoops!

Flicker chance reduced to 1%

Commit:
Checks of lights are on station

Also reduces flicker slightly

SheepFighter commented:
Sorry for not getting back sooner. The newest version is much better, but it's still far too frequent....
It might just be simpler to add a hard delay...
